api/envoy/extensions/filters/http/file_system_buffer/v3/file_system_buffer.proto

```diff
@@ -0,0 +1,114 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.file_system_buffer.v3;
+
+import "envoy/extensions/common/async_files/v3/async_file_manager.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.file_system_buffer.v3";
+option java_outer_classname = "FileSystemBufferProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/file_system_buffer/v3;file_system_bufferv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+// The behavior of the filter for a stream.
+// [#next-free-field: 6]
+message BufferBehavior {
+  oneof behavior {
+    option (validate.required) = true;
+
+    // Don't inject ``content-length`` header.
+    // Output immediately, buffer only if output is slower than input.
+    bool stream_when_possible = 1 [(validate.rules).bool = {const: true}];
+
+    // Never buffer, do nothing.
+    bool bypass = 2 [(validate.rules).bool = {const: true}];
+
+    // If ``content-length`` is not present, buffer the entire input,
+    // inject ``content-length`` header, then output.
+    // If ``content-length`` is already present, act like ``stream_when_possible``.
+    bool inject_content_length_if_necessary = 3 [(validate.rules).bool = {const: true}];
+
+    // Always buffer the entire input, and inject ``content-length``,
+    // overwriting any provided content-length header.
+    bool fully_buffer_and_always_inject_content_length = 4 [(validate.rules).bool = {const: true}];
+
+    // Always buffer the entire input, do not inject ``content-length``.
```

Comment:
What happens if a content-length was provided? Is it used?

Reply:
Changed comment to "do not modify" rather than "do not inject" to remove this ambiguity.